react-native-cli/__tests__/cli-test.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+jest.mock('fs');
+jest.mock('child_process');
+jest.dontMock('../index.js');
+jest.setMock('prompt', {
+  start: jest.genMockFn(),
+  get: jest.genMockFn()
+});
+
+describe('react-native cli', function() {
+  var originalArgs;
+  var originalLog;
+  var originalError;
+  var originalExit;
+  var originalChdir;
+
+  beforeEach(function() {
+    originalArgs = process.argv;
+    originalLog = console.log;
+    originalError = console.error;
+    originalExit = process.exit;
+    originalChdir = process.chdir;
+
+    console.log = jest.genMockFn();
+    console.error = jest.genMockFn();
+    process.exit = jest.genMockFn();
+    process.chdir = jest.genMockFn();
+  });
+
+  afterEach(function() {
+    process.argv = originalArgs;
+    console.log = originalLog;
+    console.error = originalError;
+    process.exit = originalExit;
+    process.chdir = originalChdir;
+  });
+
+  it('shows current version information and exits with "-v" flag', function() {
+    process.argv = ['node', 'react-native', '-v'];
```

Comment:
the pattern looks very repetitive and it's a bit hard to find extract the meaning of the test. How about something like this:

```javascript
expectCommandWithArgsToOutput(
  ['node', 'react-native', '-v'],
  'version:'
);
```